dev/ImageIcon/ImageIcon.cpp

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ImageIcon.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+ImageIcon::ImageIcon()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_ImageIcon);
+}
+
+void ImageIcon::OnApplyTemplate()
+{
+    Loaded({ this, &ImageIcon::OnLoaded });
+}
```

Comment:
If you call the base then by the time it comes back you should have the visual tree populated, right? If not please add a comment as to why you're waiting for Loaded.